internal/namespaces/k8s/v1beta4/custom_cluster.go

```diff
@@ -40,3 +53,50 @@ func clusterAvailableVersionsListBuilder(c *core.Command) *core.Command {
 
 	return c
 }
+
+func clusterCreateBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc(clusterActionCreate)
+	return c
+}
+
+func clusterDeleteBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc(clusterActionDelete)
+	return c
+}
+
+func clusterUpgradeBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc(clusterActionUpgrade)
+	return c
+}
+
+func clusterUpdateBuilder(c *core.Command) *core.Command {
+	c.WaitFunc = waitForClusterFunc(clusterActionUpdate)
+	return c
+}
+
+func waitForClusterFunc(action string) core.WaitFunc {
+	return func(ctx context.Context, _, respI interface{}) (interface{}, error) {
+		_, err := k8s.NewAPI(core.ExtractClient(ctx)).WaitForCluster(&k8s.WaitForClusterRequest{
+			Region:    respI.(*k8s.Cluster).Region,
+			ClusterID: respI.(*k8s.Cluster).ID,
+			Timeout:   scw.DurationPtr(clusterActionTimeout),
+		})
+		switch action {
+		case clusterActionCreate:
+			return fmt.Sprintf("Custer %s successfully created.", respI.(*k8s.Cluster).ID), nil
+		case clusterActionUpdate:
+			return fmt.Sprintf("Custer %s successfully updated.", respI.(*k8s.Cluster).ID), nil
+		case clusterActionUpgrade:
+			return fmt.Sprintf("Custer %s successfully upgraded.", respI.(*k8s.Cluster).ID), nil
+		case clusterActionDelete:
+			if err != nil {
+				notFoundError := &scw.ResourceNotFoundError{}
+				responseError := &scw.ResponseError{}
+				if errors.As(err, &responseError) && responseError.StatusCode == http.StatusNotFound || errors.As(err, &notFoundError) {
```

Comment:
Why is it a success delete if the resource is not found ? 

Shouldn't we have a proper way to notify a resource has been deleted, instead of relying on a `notFoundError || StatusNotFound` ? 

Please at least explain why this is done this way in a comment.

Reply:
Because if not, on wait for deletion, the command would return an error not found, but on deletion the error not found is a successful deletion. The not found error is taken from the TF. (I guess there isn't a nice way yet to identify it)